test/e2e-go/features/catchup/catchpointCatchup_test.go

```diff
@@ -400,6 +400,186 @@ func TestCatchpointLabelGeneration(t *testing.T) {
 	}
 }
 
+// TestReadyEndpoint starts a two-node network (configuration: `CatchpointCatchupTestNetwork.json`),
+// and execute following steps to bring up a testing scenario:
+// - set up the network with a set of consensus parameter (to allow for generating catchpoint every 4 rounds).
+// - set up primary node with config such that it generates catchpoint every 4 rounds.
+// - let network proceed for a certain rounds (in this test 17 rounds).
+// - retrieve the last catchpoint from the primary node, and let second node catch up against the catchpoint.
+//
+// The expected result would be:
+// while the second node is catching up against the catchpoint of the network,
+// `/ready` should encounter a slew of failures, then succeed with status indicating it is finishing fast-catchup.
+func TestReadyEndpoint(t *testing.T) {
+	//////////
+	// NOTE //
+	//////////
+	// This section is mostly CONSTs and helper function used in this ready endpoint test.
+	//
+	// prepareConsensus and prepareConfig tweak consensus and config to allow primary node on network
+	// `CatchpointCatchupTestNetwork.json` to generate new catchpoints over rounds.
+	const CatchpointInterval = uint64(4)
+	const CatchpointTracking = int64(2)
+	const MaxAcctLookback = uint64(2)
+	const ConfigArchival = true
+	const TargetRound = uint64(17)
+
+	prepareConsensus := func(consensusVersion protocol.ConsensusVersion) config.ConsensusProtocols {
+		consensus := make(config.ConsensusProtocols)
+		newProtocol := config.Consensus[protocol.ConsensusCurrentVersion]
+		// MaxBalLookback  =  2 x SeedRefreshInterval x SeedLookback
+		// ref. https://github.com/algorandfoundation/specs/blob/master/dev/abft.md
+		newProtocol.SeedLookback = 2
+		newProtocol.SeedRefreshInterval = 2
+		newProtocol.MaxBalLookback = 2 * newProtocol.SeedLookback * newProtocol.SeedRefreshInterval // 8
+		newProtocol.CatchpointLookback = newProtocol.MaxBalLookback
+		newProtocol.EnableOnlineAccountCatchpoints = true
+		consensus[consensusVersion] = newProtocol
+		return consensus
+	}
+
+	prepareConfig := func(a *require.Assertions, node nodecontrol.NodeController) {
+		cfg, err := config.LoadConfigFromDisk(node.GetDataDir())
+		a.NoError(err)
+		cfg.CatchpointInterval = CatchpointInterval
+		cfg.CatchpointTracking = CatchpointTracking
+		cfg.Archival = ConfigArchival
+		cfg.MaxAcctLookback = MaxAcctLookback
+		a.NoError(cfg.SaveToDisk(node.GetDataDir()))
+	}
+
```

Comment:
In case we are concerned about perf in CI test time, we can slide these lines in after we think this PR is okay, so it runs only when we want long tests.
```suggestion

	if testing.Short() {
		t.Skip()
	}

```

Reply:
NOTE: the test runs around 75s on CI, so I think we do want to run it in long test.

but there might also be some other considerations.

Reply:
is round 17 the absolute earliest we can go

Reply:
probably not, I am redesigning the test, for the `/ready` endpoint changes a bit: used to be ready when fast catchup is done, but now it is ready when sync time is 0.0 and not fast catching up.

Need to rethink a bit on how to test it, and working locally to rewrite the test.